storage/innobase/xtrabackup/test/inc/keyring_vault.sh

```diff
@@ -37,18 +27,31 @@ function keyring_vault_ping()
 	return 1
 }
 
-function keyring_vault_mount()
+function configure_keyring_file_component()
 {
-	local VAULT_CONFIG_VERSION=$1
-	local VAULT_MOUNT_VERSION=$2
 	local VAULT_MOUNT_DATA=""
-	cat > ${keyring_vault_config} <<EOF
-vault_url = ${VAULT_URL}
-secret_mount_point = ${VAULT_MOUNT_POINT}
-secret_mount_point_version = ${VAULT_MOUNT_VERSION}
-token = ${VAULT_TOKEN}
-vault_ca = ${VAULT_CA}
+  if [ "$MYSQL_DEBUG_MODE" = "on" ]; then
+    binary="mysqld-debug"
+  else
+    binary="mysqld"
+  fi
+  instance_local_manifest="${TEST_VAR_ROOT}/${binary}.my"
+  cat <<EOF > "${MYSQLD_DATADIR}/${binary}.my"
+{
+    "components": "file://component_keyring_vault"
+}
+EOF
+  cat <<EOF > "${MYSQLD_DATADIR}/component_keyring_vault.cnf"
+{
+  "vault_url": "${VAULT_URL}",
+  "secret_mount_point": "${VAULT_MOUNT_POINT}",
+  "secret_mount_point_version": "${VAULT_CONFIG_VERSION}",
+  "token": "${VAULT_TOKEN}",
+  "vault_ca": "${VAULT_CA}"
+}
 EOF
+  cp "${MYSQLD_DATADIR}/${binary}.my" ${instance_local_manifest}
+  cp "${MYSQLD_DATADIR}/component_keyring_vault.cnf" ${keyring_component_cnf}
```

Comment:
this looks like new change? this is for server to pickup vault cnf?

Reply:
Yes, this is a common set of vars on all keyring include - For example https://github.com/percona/percona-xtrabackup/blob/trunk/storage/innobase/xtrabackup/test/inc/keyring_kms.sh#L12-L14